pkg/utils/openshift_context.go

```diff
@@ -20,6 +26,36 @@ type OpenshiftContext struct {
 	OpenshiftFlavor OpenshiftFlavor
 }
 
+func NewOpenshiftContext(config *rest.Config, scheme *runtime.Scheme) (*OpenshiftContext, error) {
+	if ClusterType != ClusterTypeOpenshift {
+		return nil, nil
```

Comment:
I really don't like this. Code wise, you will have more checks elsewhere if you return a nil value and that's valid. Just return the context in a way that it conveys the fact that it is not openshift.